bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -4740,4 +4756,27 @@ public boolean equals(Object obj) {
             return this.sourceType.equals(other.sourceType) && this.targetType.equals(other.targetType);
         }
     }
+
+    /**
+     * Unordered BValue vector of size two, to hold two values being compared.
+     *
+     * @since 0.985.0
+     */
+    private static class ValuePair {
+        List<BValue> valueList = new ArrayList<>(2);
+
+        ValuePair(BValue valueOne, BValue valueTwo) {
+            valueList.add(valueOne);
+            valueList.add(valueTwo);
+        }
+
+        @Override
+        public boolean equals(Object otherPair) {
+            if (!(otherPair instanceof ValuePair)) {
+                return false;
+            }
+
+            return ((ValuePair) otherPair).valueList.containsAll(valueList);
```

Comment:
this will return true for the following case:
otherPair = [x, y]
valueList = [x', x'']

where x' and x'' are two equal values to x.

Reply:
Fixed with 91dccd1.